### ISSUE TITLE
genPostData should use a different timestamp from storageHash.oldestTimestamp when getOlderMessages =  True.

### ISSUE THREAD

**Issue by perringaiden** [IITC-CE/ingress-intel-total-conversion#364]
Currently IITC uses the stored oldest timestamp of received messages as the `maxTimestampMs` in the `data` for retrieving messages when `getOlderMessages` is `true`.

```javascript
data = $.extend(data, {maxTimestampMs: storageHash.oldestTimestamp});
```
However, where a single request results in 50+ (or a large value, unknown specific) messages all at the same time, say when a heavily layered field set is destroyed, this means that the first request sets the oldestTimestamp to that time, but every subsequent request keeps requesting the same time period.

Since the chat never receives older messages, it logjams at this timestamp and won't display any earlier entries.

**Suggested Solution:**

Use the timestamp of the oldestTimestamp minus one millisecond.  Since Niantic uses end inclusive timestamps in their request, you need to shift the end timestamp back by one millisecond or you will continue to receive the same data over and over.

```javascript
data = $.extend(data, {maxTimestampMs: storageHash.oldestTimestamp - 1});
```

**Comment by perringaiden** [comment]
Example Request:
```javascript
{"minLatE6":33837912,"minLngE6":-117942953,"maxLatE6":33905400,"maxLngE6":-117778158,"minTimestampMs":-1,"maxTimestampMs":1593087064406,"tab":"all","v":"bb473cd62691894df0733d10860034fcbeb81897"}
```
Example Response:
[InstantResponse.txt](https://github.com/IITC-CE/ingress-intel-total-conversion/files/4832540/InstantResponse.txt)

This request kept being repeated, because the max timestamp never moved earlier.

**Comment by johnd0e** [comment]
Why not open PR?

**Comment by MysticJay** [comment]
> Why not open PR?

What an effort for a two byte change...

I'll integrate it to UNIQUES040 PR as I need that anyway.

**Comment by MysticJay** [comment]
Cherry-picked the change to #389 as well

**Comment by Looka13** [comment]
Using `storageHash.oldestTimestamp - 1` as the `maxTimestampMs` would **NOT** be a correct solution, and I'll explain why:

Let's imagine that the server has _x_ > 50 messages relative to the timestamp _y_.
If we ask the server for _y_ as the `maxTimestampMs` we will receive 50 of them instead of _x_ of them (and that's how it currently works).
If instead we ask the server for _y_-1 as the `maxTimestampMs` we will receive only messages **older** then the _y_ timestamp, so all the _x_ messages will be "lost".
That would be better wrt the current situation, because it would at least let the user scroll to older posts than _y_, but it would not be the ideal solution.

In my opinion the solution should be implemented in a similar way as how the stock Intel handles the request, which is by using the `plextContinuationGuid` field in the request (with `storageHash.oldestTimestamp` as `maxTimestampMs`).
In this way the server "keeps track" of which messages (with timestamp _y_) has already sent, and replies with "different" messages each time, allowing the user to retrieve all the _x_ messages with timestamp _y_ instead of only 50 of them.

The `plextContinuationGuid` field requires the GUID of either the "first" or the "last" message the server sent (I'm sorry, I don't remember which one, I made some tests some months ago and I don't remember the answer).
The solution should look something like this:

`data = $.extend(data, {maxTimestampMs: storageHash.oldestTimestamp, plextContinuationGuid: storageHash.oldestGUID});`

Sorry if my English isn't perfect.

**Comment by Looka13** [comment]
[Ofc the solution is not complete, it would need to save in the `storageHash` the GUID of the oldest message received somewhere else in the code (probably in `window.chat.writeDataToHash`). It is just a suggestion because I'm not a programmer and thus I'm sure there exist better solutions than whichever I would ever come up with.]

**Comment by Looka13** [comment]
Sorry if I keep commenting, but I just realized that the solution proposed by @perringaiden may be worse than I initially thought.
In one of my previous messages I said that using `storageHash.oldestTimestamp - 1` as the `maxTimestampMs` would result in skipping all the messages with the same timestamp if they are more than 50.
It is not so frequent to find more than 50 messages all with the same timestamp, so that would be a somewhat rare case.
But, I just realized that the initially proposed solution affects many more cases.
For example, if a specific timestamp (let's call it _y_, as in one of my previous messages) has only two messages (which is quite common, think for example to the use of a virus, which creates as many messages with the same timestamp as the number of the resonators present on the portal), it __may__ happen that one of the two appears as last in a batch of 50 messages received from the server, and since the successive request from the client would ask _y_-1 as `maxTimestampMs`, the server would reply with messages older than _y_, skipping entirely the second of the two messages with _y_ as timestamp.

In short, using `storageHash.oldestTimestamp - 1` as the `maxTimestampMs` can potentially lead to skip many messages in the COMM.

**Comment by johnd0e** [comment]
> What an effort for a two byte change...
> 
> I'll integrate it to UNIQUES040 PR as I need that anyway.

I hope now you see your mistake (never mix up unrelated changes in single commit / PR).

**Comment by MysticJay** [comment]
@johnd0e Well, it was not done that way....

**Comment by johnd0e** [comment]
> I hope now you see your mistake (never mix up unrelated changes in single commit / PR).

I mean cbb38684b0122d18ae7a9a2c8d136d4bb4844a62 of #389

**Comment by MysticJay** [comment]
@johnd0e I don't remember who proposed it. The changes are very small. A direct commit to master was not wanted, single PRs cause overhead. Creating one commit per file was a reasonable option. The PR should be merged by cherry-picking each commit (which have been crafted to only effect one file at a time). But we can certainly split them up again in PRs with all the overhead for a 2-byte change.

There is always a dark side to everything!

**Comment by johnd0e** [comment]
> Creating one commit per file was a reasonable option.

Wrong.
One set of related changes per commit.
Whatever number of files.
That is in our wiki.

> But we can certainly split them up again in PRs with all the overhead for a 2-byte change.

That's exactly what I'd suggest you to prefer next time.

**Comment by johnd0e** [comment]
@Looka13

Thank you for this investigation!

**Comment by Looka13** [comment]
@johnd0e Thank **you** and all the other developers who works on such an important tool as it is IITC!
